salt/modules/lxc.py

```diff
@@ -2294,83 +2374,73 @@ def update_lxc_conf(name, lxc_conf, lxc_conf_unset):
 
     changes = {'edited': [], 'added': [], 'removed': []}
     ret = {'changes': changes, 'result': True, 'comment': ''}
-    lxc_conf_p = '/var/lib/lxc/{0}/config'.format(name)
-    if not exists(name):
-        ret['result'] = False
-        ret['comment'] = 'Container does not exist: {0}'.format(name)
-    elif not os.path.exists(lxc_conf_p):
-        ret['result'] = False
-        ret['comment'] = (
-            'Configuration does not exist: {0}'.format(lxc_conf_p))
-    else:
-        with salt.utils.fopen(lxc_conf_p, 'r') as fic:
-            filtered_lxc_conf = []
-            for row in lxc_conf:
-                if not row:
-                    continue
-                for conf in row:
-                    try:
-                        filtered_lxc_conf.append((conf.strip(),
-                                                  row[conf].strip()))
-                    except AttributeError:
-                        filtered_lxc_conf.append((conf.strip(),
-                                                  str(row[conf]).strip()))
-            ret['comment'] = 'lxc.conf is up to date'
-            lines = []
-            orig_config = fic.read()
-            for line in orig_config.splitlines():
-                if line.startswith('#') or not line.strip():
-                    lines.append([line, ''])
-                else:
-                    line = line.split('=')
-                    index = line.pop(0)
-                    val = (index.strip(), '='.join(line).strip())
-                    if val not in lines:
-                        lines.append(val)
-            for k, item in filtered_lxc_conf:
-                matched = False
-                for idx, line in enumerate(lines[:]):
-                    if line[0] == k:
-                        matched = True
-                        lines[idx] = (k, item)
-                        if '='.join(line[1:]).strip() != item.strip():
-                            changes['edited'].append(
-                                ({line[0]: line[1:]}, {k: item}))
-                            break
-                if not matched:
-                    if not (k, item) in lines:
-                        lines.append((k, item))
-                    changes['added'].append({k: item})
-            dest_lxc_conf = []
-            # filter unset
-            if lxc_conf_unset:
-                for line in lines:
-                    for opt in lxc_conf_unset:
-                        if (
-                            not line[0].startswith(opt)
-                            and line not in dest_lxc_conf
-                        ):
-                            dest_lxc_conf.append(line)
-                        else:
-                            changes['removed'].append(opt)
+
+    # do not use salt.utils.fopen !
+    with open(lxc_conf_p, 'r') as fic:
```

Comment:
you are free to let my fork diverges and let the official lxc module broken, no problem at all with that :)